File: src/main/java/com/lucasbezerra/cursomc/services/CategoriaService.java
package com.lucasbezerra.cursomc.services;

import com.lucasbezerra.cursomc.damain.Categoria;
import com.lucasbezerra.cursomc.repositories.CategoriaRepository;
import com.lucasbezerra.cursomc.services.exceptions.ObjectNotFoundException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class CategoriaService {

    @Autowired
    private CategoriaRepository repo;

    public Categoria buscar(Integer id){
        Optional<Categoria> obj = repo.findById(id);
        return obj.orElseThrow(() -> new ObjectNotFoundException("Objeto não encontrado! Id: "+ id + ", Tipo: " + Categoria.class.getName()));
    }
}

File: src/main/kotlin/com/bzrra/cursomc/services/CategoriaService.kt
package com.bzrra.cursomc.services

import com.bzrra.cursomc.domain.Categoria
import com.bzrra.cursomc.repositories.CategoriaRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Example
import org.springframework.stereotype.Service
import java.util.*

@Service
class CategoriaService {

    @Autowired
    private lateinit var repo: CategoriaRepository

    fun buscar(id: Int): Categoria? {
        val obj: Optional<Categoria> = repo.findById(id)

        return obj.orElse(null)
    }
}